src/anomalib/models/video/fuvas/lightning_model.py

```diff
@@ -0,0 +1,200 @@
+    Anomaly detection is based on feature reconstruction error.
+
+See Also:
+    :class:`anomalib.models.video.fuvas.torch_model.FUVASModel`:
+        PyTorch implementation of the FUVAS model.
+"""
+
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from typing import Any
+
+import torch
+from lightning.pytorch.utilities.types import STEP_OUTPUT
+
+from anomalib import LearningType
+from anomalib.data import Batch
+from anomalib.metrics import Evaluator
+from anomalib.models.components import AnomalibModule, MemoryBankMixin
+from anomalib.post_processing import PostProcessor
+from anomalib.pre_processing import PreProcessor
+from anomalib.visualization import Visualizer
+
+from .torch_model import FUVASModel
+
+logger = logging.getLogger(__name__)
+
+
+class Fuvas(MemoryBankMixin, AnomalibModule):
+    """FUVAS Lightning Module.
+
+    Args:
+        backbone (str): Name of the backbone 3D CNN/transformer network.
+            Defaults to ``"x3d_s"``.
+        layer (str): Name of the layer to extract features from the backbone.
+            Defaults to ``"blocks.4"``.
+        pre_trained (bool, optional): Whether to use a pre-trained backbone.
+            Defaults to ``True``.
+        spatial_pool (bool, optional): Whether to use spatial pooling on features.
+            Defaults to ``True``.
+        pooling_kernel_size (int, optional): Kernel size for pooling features.
+            Defaults to ``1``.
+        pca_level (float, optional): Ratio of variance to preserve in PCA.
+            Must be between 0 and 1.
+            Defaults to ``0.98``.
+        do_seg (bool, optional): Whether to perform anomaly segmentation.
+            If False, only detection scores are computed.
+            Defaults to ``True``.
+        pre_processor (PreProcessor | bool, optional): Pre-processor to use.
+            If ``True``, uses the default pre-processor.
+            If ``False``, no pre-processing is performed.
+            Defaults to ``True``.
+        post_processor (PostProcessor | bool, optional): Post-processor to use.
+            If ``True``, uses the default post-processor.
+            If ``False``, no post-processing is performed.
+            Defaults to ``True``.
+        evaluator (Evaluator | bool, optional): Evaluator to use.
+            If ``True``, uses the default evaluator.
+            If ``False``, no evaluation is performed.
+            Defaults to ``True``.
+        visualizer (Visualizer | bool, optional): Visualizer to use.
+            If ``True``, uses the default visualizer.
+            If ``False``, no visualization is performed.
+            Defaults to ``True``.
+    """
+
+    def __init__(
+        self,
+        backbone: str = "x3d_s",
+        layer: str = "blocks.4",
+        pre_trained: bool = True,
+        spatial_pool: bool = True,
+        pooling_kernel_size: int = 1,
+        pca_level: float = 0.98,
+        do_seg: bool = True,
```

Comment:
Do we always need this `do_seg` argument?

Reply:
If you really need this, I think you could use anomalib's `TaskType` enum from here
https://github.com/open-edge-platform/anomalib/blob/7a3bf3aa9ba324b8a76e34606b0cf1e45e283e0a/src/anomalib/__init__.py#L65-L88

Reply:
so your argument would be `task: TaskType | str = "segmentation"`

Reply:
@djdameln, can we approach this differently to follow the new approach in dataclasses and metrics?